drivers/ipvlan/ipvlan_setup.go

```diff
@@ -0,0 +1,204 @@
+		return "", fmt.Errorf("Failed to create the ipvlan link: %v", err)
+	}
+	return ipvlan.Attrs().Name, nil
+}
+
+// setIPVlanMode setter for one of the two ipvlan port types
+func setIPVlanMode(mode string) (netlink.IPVlanMode, error) {
+	switch mode {
+	case modeL2:
+		return netlink.IPVLAN_MODE_L2, nil
+	case modeL3:
+		return netlink.IPVLAN_MODE_L3, nil
+	default:
+		return 0, fmt.Errorf("Unknown ipvlan mode: %s", mode)
+	}
+}
+
+// validateHostIface check if the specified interface exists in the default namespace
+func validateHostIface(ifaceStr string) bool {
+	_, err := netlink.LinkByName(ifaceStr)
+	if err != nil {
+		return false
+	}
+	return true
+}
+
+// kernelSupport for the necessary kernel module for the driver type
+func kernelSupport(networkTpe string) error {
+	// attempt to load the module, silent if successful or already loaded
+	exec.Command("modprobe", ipvlanType).Run()
+	f, err := os.Open("/proc/modules")
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	s := bufio.NewScanner(f)
+	for s.Scan() {
+		if strings.Contains(s.Text(), ipvlanType) {
+			return nil
+		}
+	}
+	return fmt.Errorf("required kernel module '%s' was not found in /proc/modules, ipvlan requires kernel version >= 3.19", ipvlanType)
+}
+
+// validateSubIface verifies the -o host_iface=eth0.10 (vlan 10) if it does not exist it infers a vlan/subinterface
+func validateSubIface(ifaceName string) error {
+	if strings.Contains(ifaceName, ".") {
+		// parse --host_iface=eth0.10
+		splitIface := strings.Split(ifaceName, ".")
+		if len(splitIface) != 2 {
+			return fmt.Errorf("required -o host_iface format is eth_name.vlan_id (tagged Ex. eth0.10 for vlan 10) or an existing interface (untagged Ex. eth0) instead received %s", ifaceName)
+		}
+		parentIface, vidStr := splitIface[0], splitIface[1]
+		// validate type and convert vlan id to int
+		vid, err := strconv.Atoi(vidStr)
+		if err != nil {
+			return fmt.Errorf("unable to parse a valid parent iface: %s and vlan id from: %s (Ex. eth0.10 for vlan 10)", parentIface, vidStr)
+		}
+		// VLAN identifier or VID is a 12-bit field specifying the VLAN to which the frame belongs
+		if vid > 4094 {
+			return fmt.Errorf("vlan id must be between 1-4094, received %d", vid)
+		}
+		if vid < 1 {
+			return fmt.Errorf("vlan id must be between 1-4094, received %d", vid)
+		}
+		// Check if the interface exists
+		if ok := validateHostIface(parentIface); !ok {
+			return fmt.Errorf("-o host_iface parent interface does was not found on the host: %s", parentIface)
+		}
+	} else {
+		return fmt.Errorf("required -o host_iface format is eth_name.vlan_id (tagged Ex. eth0.10 for vlan 10) or an existing interface (untagged Ex. eth0) instead received %s", ifaceName)
+	}
+	return nil
+}
+
+// createVlanSubIface parses sub-interfaces and vlan id for creation
+func createVlanSubIface(ifaceName string) error {
+	if strings.Contains(ifaceName, ".") {
```

Comment:
Duplicated checks, we have already doing this in `validateSubIface`. I think we can consolidate these two functions. Maybe rename it to ensureSubIface?